src/vaults/yields.utils.ts

```diff
@@ -523,35 +535,40 @@ async function evaluateYieldEvents(chain: Chain, vault: VaultDefinitionModel): P
  * @returns yield sources estimating the aggregate performance
  */
 export async function queryVaultYieldSources(chain: Chain, vault: VaultDefinitionModel): Promise<CachedYieldSource[]> {
-  const { compoundApr, compoundApy, grossCompoundApr, grossCompoundApy, tokenEmissionAprs } = await evaluateYieldEvents(
-    chain,
-    vault,
-  );
-
-  const compoundSources = [];
-
-  if (compoundApr > 0) {
-    // create the apr source for harvests
-    const compoundYieldSource = createYieldSource(
-      vault,
-      SourceType.PreCompound,
-      VAULT_SOURCE,
-      compoundApr,
-      grossCompoundApr,
-    );
-    compoundSources.push(compoundYieldSource);
+  try {
+    const { compoundApr, compoundApy, grossCompoundApr, grossCompoundApy, tokenEmissionAprs } =
+      await evaluateYieldEvents(chain, vault);
+
+    const compoundSources = [];
+
+    if (compoundApr > 0) {
```

Comment:
apy is calculated from this right? there would be no need to check compoundApr and compoundApy because they will either both have or both not have values?

Reply:
correct